Generate interpolation wrapper map (formerly the CSS animation property wrapper map)

weinig opened the pull request:
#### 0dce5e774ce4877e4170b53e3ff5dff96b9d083f
<pre>
Generate interpolation wrapper map (formerly the CSS animation property wrapper map)
<a href="https://bugs.webkit.org/show_bug.cgi?id=287878">https://bugs.webkit.org/show_bug.cgi?id=287878</a>

Reviewed by Antti Koivisto and Darin Adler.

Moves generation of the interpolation wrapper map, formerly called the
CSS animation property wrapper map, to be generated from details in
CSSProperties.json. The main benefit right now, other than having the
details localized, is that this automatically allows generation of the
correct shorthand wrappers based on which longhands have wrappers.

To go with the generation, the wrapper and blend functions have been split
into their own files, and they have all been moved into a new namespace:
`Style::Interpolation`. This is motivated by:
  - the code operates on post-style building data structures, which now
    use the Style prefix/namespacing.
  - the code will be used for more than just animation in the future
    with the introduction of explicit means to interpolate from css.

* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Headers.cmake:
* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
    - Add/rename files.

* Source/WebCore/css/process-css-properties.py:
* Source/WebCore/css/CSSProperties.json:
    - Adds generation of the wrapper map from new animation-wrapper-* properties
      added to CSSProperties.json.

* Source/WebCore/rendering/style/RenderStyle.cpp:
* Source/WebCore/rendering/style/RenderStyle.h:
* Source/WebCore/rendering/style/RenderStyleInlines.h:
    - Renamed border image/mask slice getters/setter to match the property name and
      sibling properties, removing the trailing &apos;s&apos;.

* Source/WebCore/rendering/style/ShadowData.h:
    - Move function from CSSPropertyAnimation.cpp so it can be used by multiple files.

* Source/WebCore/platform/Length.h:
    - Move function from CSSPropertyAnimation.cpp so it can be used by multiple files.

* Source/WebCore/style/StyleInterpolationClient.h: Renamed from Source/WebCore/animation/CSSPropertyBlendingClient.h.
    - Rename CSSPropertyBlendingClient to Style::Interpolation::Client, to match new naming.

* Source/WebCore/style/StyleInterpolation.cpp: Added.
* Source/WebCore/style/StyleInterpolation.h: Added.
* Source/WebCore/style/StyleInterpolationContext.h: Copied from Source/WebCore/animation/CSSPropertyAnimation.h.
* Source/WebCore/style/StyleInterpolationFunctions.h: Added.
* Source/WebCore/style/StyleInterpolationWrapperBase.h: Renamed from Source/WebCore/animation/CSSPropertyAnimation.h.
* Source/WebCore/style/StyleInterpolationWrappers.h: Added.
    - Split CSSPropertyAnimation.cpp up into a few files to make it a bit more manageable.
        The exported functions go to StyleInterpolation.h/cpp
        The base wrapper goes to StyleInterpolationWrapperBase.h
        The concrete wrappers go to StyleInterpolationWrappers.h
        The blend functions go to StyleInterpolationFunctions.
        The client goes to CSSPropertyBlendingClient.h
        The blending context goes to StyleInterpolationContext.h

* Source/WebCore/style/Styleable.cpp:
* Source/WebCore/animation/BlendingKeyframes.cpp:
* Source/WebCore/animation/CSSPropertyAnimation.cpp: Removed.
* Source/WebCore/animation/KeyframeEffect.cpp:
* Source/WebCore/animation/KeyframeEffect.h:
* Source/WebCore/animation/KeyframeEffectStack.cpp:
* Source/WebCore/animation/KeyframeInterpolation.cpp:
* Source/WebCore/animation/KeyframeInterpolation.h:
* Source/WebCore/animation/WebAnimation.cpp:
* Source/WebCore/css/CSSComputedStyleDeclaration.cpp:
* Source/WebCore/css/CSSProperty.h:
* Source/WebCore/css/ComputedStyleExtractor.cpp:
* Source/WebCore/platform/animation/AcceleratedEffect.cpp:
    - Update for renames.

* Tools/Scripts/webkitpy/style/checkers/jsonchecker.py:
    - Update for new animation-wrapper-* properties in CSSProperties.json.

Canonical link: <a href="https://commits.webkit.org/291227@main">https://commits.webkit.org/291227@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7b1df5922d929a493adf2dd524b8f49b132d260e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/92287 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/11821 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/1379 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/97290 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/42812 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/94337 "Passed tests") | [❌ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/12122 "Hash 7b1df592 for PR 41106 does not build (failure)") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/20291 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/70756 "Found 2 new test failures: imported/w3c/web-platform-tests/html/semantics/forms/constraints/infinite_backtracking.html imported/w3c/web-platform-tests/service-workers/service-worker/redirected-response.https.html (failure)") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/28210 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/95288 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/130/builds/12122 "Hash 7b1df592 for PR 41106 does not build (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/83561 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/51084 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/91782 "Passed tests") | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/130/builds/12122 "Hash 7b1df592 for PR 41106 does not build (failure)") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/1181 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/42144 "Built successfully") | 
| | [❌ 🧪 api-ios](https://ews-build.webkit.org/#/builders/130/builds/12122 "Hash 7b1df592 for PR 41106 does not build (failure)") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/1143 "") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/99313 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/19355 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/14317 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/79792 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/19606 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/79428 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/79040 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/19592 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/23563 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/860 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/12356 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/19337 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/24507 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/19029 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/22486 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/20768 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rniwa commented on Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
```diff
@@ -1313,6 +1312,7 @@ style/StyleBuilder.cpp
 style/StyleBuilderConverter.h
 style/StyleBuilderCustom.h
 style/StyleBuilderState.cpp
+style/StyleInterpolation.cpp
```
Would you mind adding StyleInterpolationFunctions.h and StyleInterpolationWrappers.h here as well?
That'll keep the safer C++ bot green.

rniwa commented on Source/WebCore/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations:
```diff
@@ -136,7 +136,6 @@ animation/AnimationTimeline.cpp
 animation/AnimationTimelinesController.cpp
 animation/BlendingKeyframes.cpp
 animation/CSSAnimation.cpp
-animation/CSSPropertyAnimation.cpp
```
Ditto. Can we add these two files as expected failures here as well?

darinadler commented on Source/WebCore/animation/KeyframeEffect.cpp:
```diff
@@ -1860,7 +1860,7 @@ void KeyframeEffect::setAnimatedPropertiesInStyle(RenderStyle& targetStyle, cons
     if (m_animationType == WebAnimationType::CSSTransition) {
         ASSERT(properties.size() == 1);
-        CSSPropertyAnimation::blendProperty(*this, *properties.begin(), targetStyle, *m_blendingKeyframes[0].style(), *m_blendingKeyframes[1].style(), iterationProgress, m_compositeOperation);
+        Style::Interpolation::interpolate(*properties.begin(), targetStyle, *m_blendingKeyframes[0].style(), *m_blendingKeyframes[1].style(), iterationProgress, m_compositeOperation, IterationCompositeOperation::Replace, 0, *this);
```
Not thrilled with the magic number 0 here. Should we have an overload for this case to make it easier to read at the call site if this `Replace, 0` flavor could be a named function?

darinadler commented on Source/WebCore/style/StyleInterpolationWrapperBase.h:
```diff
@@ -28,32 +30,45 @@
-#include "IterationCompositeOperation.h"
-#include "WebAnimationTypes.h"
-#include <wtf/HashSet.h>
+// NOTE: DO NOT INCLUDE THIS FILE ANYWHERE EXCEPT FROM GENERATED CODE.
```
Can we check this with a macro instead of suggesting it with a comment?

weinig replied:
Oh yeah, I think we can. I'll do that. Good idea.

darinadler commented on Source/WebCore/style/StyleInterpolationWrappers.h:
```diff
@@ -0,0 +1,2774 @@
+class LengthBoxWrapper : public WrapperWithGetter<const LengthBox&> {
+    WTF_MAKE_FAST_ALLOCATED_WITH_HEAP_IDENTIFIER(Animation);
+public:
+    enum class Flags {
```
Consider `: uint8_t`?

darinadler commented on Source/WebCore/style/StyleInterpolationWrappers.h:
```diff
@@ -0,0 +1,2774 @@
+            return true;
+
+        const ShadowData* shadowA = (a.*m_getter)();
+        const ShadowData* shadowB = (b.*m_getter)();
```
```suggestion
        auto* shadowA = (a.*m_getter)();
        auto* shadowB = (b.*m_getter)();
```

darinadler commented on Source/WebCore/style/StyleInterpolationWrappers.h:
```diff
@@ -0,0 +1,2774 @@
+
+            // end of just one of the lists
+            if (!shadowA || !shadowB)
+                return false;
```
```suggestion
            if (!shadowA || !shadowB)
                return shadowA == shadowB;
```

darinadler commented on Source/WebCore/style/StyleInterpolationWrappers.h:
```diff
@@ -0,0 +1,2774 @@
+            return true;
+
+        const ShadowData* fromShadow = (from.*m_getter)();
+        const ShadowData* toShadow = (to.*m_getter)();
```
```suggestion
        auto* fromShadow = (from.*m_getter)();
        auto* toShadow = (to.*m_getter)();
```

darinadler commented on Source/WebCore/style/StyleInterpolationWrappers.h:
```diff
@@ -0,0 +1,2774 @@
+        }
+
+        int fromLength = shadowListLength(fromShadow);
+        int toLength = shadowListLength(toShadow);
```
```suggestion
        size_t fromLength = shadowListLength(fromShadow);
        size_t toLength = shadowListLength(toShadow);
```

weinig replied:
Why the explicit size_t here instead of auto?

darinadler replied:
I'd be fine with `auto`.

darinadler commented on Source/WebCore/style/StyleInterpolationWrappers.h:
```diff
@@ -0,0 +1,2774 @@
+    void interpolate(RenderStyle& destination, const RenderStyle& from, const RenderStyle& to, const Context& context) const final
+    {
+        const ShadowData* fromShadow = (from.*m_getter)();
+        const ShadowData* toShadow = (to.*m_getter)();
```
```suggestion
        auto* fromShadow = (from.*m_getter)();
        auto* toShadow = (to.*m_getter)();
```

darinadler commented on Source/WebCore/style/StyleInterpolationWrappers.h:
```diff
@@ -0,0 +1,2774 @@
+
+        while (shadowA || shadowB) {
+            const ShadowData* srcShadow = shadowForInterpolation(shadowA, shadowB);
+            const ShadowData* dstShadow = shadowForInterpolation(shadowB, shadowA);
```
```suggestion
            auto* srcShadow = shadowForInterpolation(shadowA, shadowB);
            auto* dstShadow = shadowForInterpolation(shadowB, shadowA);
```

darinadler commented on Source/WebCore/style/StyleInterpolationWrappers.h:
```diff
@@ -0,0 +1,2774 @@
+        auto addShadows = [&](const ShadowData* shadow) {
+            while (shadow) {
+                auto blendedShadow = makeUnique<ShadowData>(*shadow);
+                auto* blendedShadowPtr = blendedShadow.get();
+                if (!lastShadow)
+                    newShadowData = WTFMove(blendedShadow);
+                else
+                    lastShadow->setNext(WTFMove(blendedShadow));
+
+                lastShadow = blendedShadowPtr;
```
Consider a helper function to do this append operation, which is repeated below in `blendSimpleOrMatchedShadowLists`.

darinadler commented on Source/WebCore/style/StyleInterpolationWrappers.h:
```diff
@@ -0,0 +1,2774 @@
+            const ShadowData* srcShadow = shadowForInterpolation(shadowA, shadowB);
+            const ShadowData* dstShadow = shadowForInterpolation(shadowB, shadowA);
+
+            std::unique_ptr<ShadowData> blendedShadow = blendFunc(srcShadow, dstShadow, styleA, styleB, context);
```
```suggestion
            auto blendedShadow = blendFunc(srcShadow, dstShadow, styleA, styleB, context);
```

darinadler commented on Source/WebCore/style/StyleInterpolationWrappers.h:
```diff
@@ -0,0 +1,2774 @@
+            const ShadowData* srcShadow = shadowForInterpolation(fromShadow, toShadow);
+            const ShadowData* dstShadow = shadowForInterpolation(toShadow, fromShadow);
+
+            std::unique_ptr<ShadowData> blendedShadow = blendFunc(srcShadow, dstShadow, styleA, styleB, context);
```
```suggestion
            auto* fromShadow = i < fromLength ? fromShadows[i] : nullptr;
            auto* toShadow = i < toLength ? toShadows[i] : nullptr;

            auto* srcShadow = shadowForInterpolation(fromShadow, toShadow);
            auto* dstShadow = shadowForInterpolation(toShadow, fromShadow);

            auto blendedShadow = blendFunc(srcShadow, dstShadow, styleA, styleB, context);
```

darinadler commented on Source/WebCore/style/StyleInterpolationFunctions.h:
```diff
@@ -0,0 +1,703 @@
+}
+
+
+inline bool lengthVariantRequiresInterpolationForAccumulativeIteration(const LengthSize& from, const LengthSize& to)
```
```suggestion
}

inline bool lengthVariantRequiresInterpolationForAccumulativeIteration(const LengthSize& from, const LengthSize& to)
```

darinadler commented on Source/WebCore/style/StyleInterpolationFunctions.h:
```diff
@@ -0,0 +1,703 @@
+        return context.progress ? to : from;
+    }
+
+    ASSERT(from && to);
```
```suggestion
    ASSERT(from);
    ASSERT(to);
```

darinadler commented on Source/WebCore/style/StyleInterpolationFunctions.h:
```diff
@@ -0,0 +1,703 @@
+        return context.progress ? to : from;
+    }
+
+    ASSERT(from && to);
```
```suggestion
    ASSERT(from);
    ASSERT(to);
```

darinadler commented on Source/WebCore/style/StyleInterpolationFunctions.h:
```diff
@@ -0,0 +1,703 @@
+        float interpolated = blendFunc(fromItem.value(), toItem.value(), context);
+        result.insert({ fromItem.tag(), interpolated });
+    }
+    return result;
```
```suggestion
    ASSERT(from.size() == to.size());
    FontVariationSettings result;
    size_t size = from.size();
    for (size_t i = 0; i < size; ++i) {
        auto& fromItem = from.at(i);
        auto& toItem = to.at(i);
        ASSERT(fromItem.tag() == toItem.tag());
        result.insert({ fromItem.tag(), blendFunc(fromItem.value(), toItem.value(), context) });
    }
    return result;
```

darinadler commented on Source/WebCore/style/StyleInterpolationFunctions.h:
```diff
@@ -0,0 +1,703 @@
+    }, [&](const GridTrackEntryRepeat& repeat) {
+        if (!std::holds_alternative<GridTrackEntryRepeat>(to.list[i]))
+            return false;
+        const auto& toEntry = std::get<GridTrackEntryRepeat>(to.list[i]);
```
```suggestion
       auto& toEntry = std::get<GridTrackEntryRepeat>(to.list[i]);
```

darinadler commented on Source/WebCore/style/StyleInterpolationFunctions.h:
```diff
@@ -0,0 +1,703 @@
+        else {
+            const Vector<String>& toNames = std::get<Vector<String>>(to[i]);
+            result.append(toNames);
+        }
```
```suggestion
        else
            result.append(std::get<Vector<String>>(to[i]));
```

darinadler commented on Source/WebCore/style/StyleInterpolationFunctions.h:
```diff
@@ -0,0 +1,703 @@
+{
+    if (context.isDiscrete)
+        return context.progress < 0.5 ? from : to;
+    ASSERT(from && to);
```
```suggestion
    ASSERT(from);
    ASSERT(to);
```

darinadler commented on Source/WebCore/style/StyleInterpolation.cpp:
```diff
@@ -0,0 +1,386 @@
+    if (from.values.size() != to.values.size())
+        return std::nullopt;
+
+    Vector<CSSCustomPropertyValue::SyntaxValue> interpolatedSyntaxValues;
```
Do we want to reserve initial capacity?

weinig replied:
What this instead:

```c++
    auto interpolatedSyntaxValues = WTF::compactMap(zippedRange(from.values, to.values), [&](const auto& pair) {
        return interpolateSyntaxValues(fromStyle, toStyle, pair.first, pair.second, context);
    });
```

weinig replied:
Alas, it doesn't compile, error in CollectionInspector. Not going to investigate that now, but would be good make that kind of thing work. 

I did keep the zippedRange part though.

darinadler commented on Source/WebCore/style/StyleInterpolation.cpp:
```diff
@@ -0,0 +1,386 @@
+            return propertyId == CSSPropertyCustom || !!WrapperMap::singleton().wrapper(propertyId);
+        },
+        [](const AtomString&) {
+            // FIXME: this should only be true for property that are registered custom properties.
```
```suggestion
            // FIXME: This should only be true for properties that are registered custom properties.
```
But also, I’m not sure I understand the meaning?

weinig replied:
Me either. (to be clear, its an existing comment)

darinadler commented on Source/WebCore/style/StyleInterpolation.cpp:
```diff
@@ -0,0 +1,386 @@
+                            return false;
+                    }
+                    return true;
+                }
```
```suggestion
                if (fromSyntaxValues.size() != toSyntaxValues.size())
                    return false;
                for (size_t i = 0; i < fromSyntaxValues.size(); ++i) {
                    if (!syntaxValuesRequireInterpolationForAccumulativeIteration(fromSyntaxValues[i], toSyntaxValues[i], true))
                        return false;
                }
                return true;
```

weinig committed:
Generate interpolation wrapper map (formerly the CSS animation property wrapper map)
https://bugs.webkit.org/show_bug.cgi?id=287878

Reviewed by Antti Koivisto and Darin Adler.

Moves generation of the interpolation wrapper map, formerly called the
CSS animation property wrapper map, to be generated from details in
CSSProperties.json. The main benefit right now, other than having the
details localized, is that this automatically allows generation of the
correct shorthand wrappers based on which longhands have wrappers.

To go with the generation, the wrapper and blend functions have been split
into their own files, and they have all been moved into a new namespace:
`Style::Interpolation`. This is motivated by:
  - the code operates on post-style building data structures, which now
    use the Style prefix/namespacing.
  - the code will be used for more than just animation in the future
    with the introduction of explicit means to interpolate from css.

* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Headers.cmake:
* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
    - Add/rename files.

* Source/WebCore/css/process-css-properties.py:
* Source/WebCore/css/CSSProperties.json:
    - Adds generation of the wrapper map from new animation-wrapper-* properties
      added to CSSProperties.json.

* Source/WebCore/rendering/style/RenderStyle.cpp:
* Source/WebCore/rendering/style/RenderStyle.h:
* Source/WebCore/rendering/style/RenderStyleInlines.h:
    - Renamed border image/mask slice getters/setter to match the property name and
      sibling properties, removing the trailing 's'.

* Source/WebCore/rendering/style/ShadowData.h:
    - Move function from CSSPropertyAnimation.cpp so it can be used by multiple files.

* Source/WebCore/platform/Length.h:
    - Move function from CSSPropertyAnimation.cpp so it can be used by multiple files.

* Source/WebCore/style/StyleInterpolationClient.h: Renamed from Source/WebCore/animation/CSSPropertyBlendingClient.h.
    - Rename CSSPropertyBlendingClient to Style::Interpolation::Client, to match new naming.

* Source/WebCore/style/StyleInterpolation.cpp: Added.
* Source/WebCore/style/StyleInterpolation.h: Added.
* Source/WebCore/style/StyleInterpolationContext.h: Copied from Source/WebCore/animation/CSSPropertyAnimation.h.
* Source/WebCore/style/StyleInterpolationFunctions.h: Added.
* Source/WebCore/style/StyleInterpolationWrapperBase.h: Renamed from Source/WebCore/animation/CSSPropertyAnimation.h.
* Source/WebCore/style/StyleInterpolationWrappers.h: Added.
    - Split CSSPropertyAnimation.cpp up into a few files to make it a bit more manageable.
        The exported functions go to StyleInterpolation.h/cpp
        The base wrapper goes to StyleInterpolationWrapperBase.h
        The concrete wrappers go to StyleInterpolationWrappers.h
        The blend functions go to StyleInterpolationFunctions.
        The client goes to CSSPropertyBlendingClient.h
        The blending context goes to StyleInterpolationContext.h

* Source/WebCore/style/Styleable.cpp:
* Source/WebCore/animation/BlendingKeyframes.cpp:
* Source/WebCore/animation/CSSPropertyAnimation.cpp: Removed.
* Source/WebCore/animation/KeyframeEffect.cpp:
* Source/WebCore/animation/KeyframeEffect.h:
* Source/WebCore/animation/KeyframeEffectStack.cpp:
* Source/WebCore/animation/KeyframeInterpolation.cpp:
* Source/WebCore/animation/KeyframeInterpolation.h:
* Source/WebCore/animation/WebAnimation.cpp:
* Source/WebCore/css/CSSComputedStyleDeclaration.cpp:
* Source/WebCore/css/CSSProperty.h:
* Source/WebCore/css/ComputedStyleExtractor.cpp:
* Source/WebCore/platform/animation/AcceleratedEffect.cpp:
    - Update for renames.

* Tools/Scripts/webkitpy/style/checkers/jsonchecker.py:
    - Update for new animation-wrapper-* properties in CSSProperties.json.

Canonical link: https://commits.webkit.org/291227@main
